src/manage.c

```diff
@@ -5995,6 +6025,50 @@ feed_lockfile_lock (lockfile_t *lockfile)
   return 0;
 }
 
+/**
+ * @brief Acquires the feed lock and writes the current time to the lockfile.
+ *
+ * @param[out] lockfile   Lockfile data struct.
+ *
+ * @return 0 success, 1 already locked, -1 error
+ */
+int
+feed_lockfile_lock_timeout (lockfile_t *lockfile)
+{
+  int ret;
+  gboolean log_timeout;
+  time_t timeout_end;
+
+  /* Try to lock the file */
+
+  log_timeout = TRUE;
+  timeout_end = time (NULL) + feed_lock_timeout;
+  do
+    {
+      ret = lockfile_lock_path_nb (lockfile, get_feed_lock_path ());
+      if (ret == 1 && timeout_end > time (NULL))
+        {
+          if (log_timeout)
+            {
+              log_timeout = FALSE;
+              g_message ("%s: Feed is currently locked by another process,"
+                         " will retry until %s.",
+                         __func__, iso_time (&timeout_end));
+            }
+          gvm_sleep (1);
+        }
+      else if (ret)
+        {
+          return ret;
+        }
+    } while (ret);
```

Comment:
What's the meaning of ret here? 1 == locked? If yes maybe you could rename it to somthing like is_locked.

Reply:
I've renamed the variable and added a few comments about the conditions.